packages/elements/src/interactive-chart/index.ts

```diff
@@ -873,10 +830,17 @@ export class InteractiveChart extends ResponsiveElement {
         span.textContent = value as string;
         rowLegendElem[index].appendChild(span);
       } else {
-        this.createTextOHLC(rowLegendElem, value as BarData, priceColor, index);
+        this.createTextOHLC(rowLegendElem, value as OhlcData, priceColor, index);
       }
     } else {
-      this.createTextPrice(rowLegendElem, value as number | string, priceColor, index);
+      let price: string | number;
+      if ((value as SingleValueData).value !== undefined) {
+        price = (value as SingleValueData).value;
+      } else {
+        price = value as number;
+      }
```

Comment:
this should be simpler.
```suggestion
      const subValue = (value as SingleValueData).value;
      const price = subValue !== undefined ? subValue : value as number; 
```

Reply:
How about this one?
```suggestion
     const price = (value as SingleValueData).value || value as number; 
```

Reply:
if `value.value` is `0`, `price` would set to `value` instead of `value.value`. This changes code behavior so we better avoid this.

Reply:
Then how about this one. 😁

```suggestion
     const price = (value as SingleValueData).value ?? value; 
```

Reply:
That should work! Don't forget to check the type of `price` tho.